### ISSUE TITLE
Low output after merging (16S)

### ISSUE THREAD

**Issue by natasharg** [benjjneb/dada2#761]
Hi,

I am fairly new to using R and I am working on a project that requires sequencing of 16S and ITS regions. I am working on 16S right now (515F/806R primers) and I am getting low to no output of reads after merging/filtering. I am following the DADA2 Pipeline Tutorial (1.12). This is the summary of the number of reads through each step:
           input filtered denoisedF denoisedR merged nonchim
Sample-1  29112    27209     23925     19443      0       0
Sample-2  19059    17868     16155     12670      0       0
Sample-3 99195    94597     91753     89818     78      78
Sample-4 76890    73785     71592     67441     14      14
Sample-5   29173    28019     25746     24773      0       0
Sample-6   26079    25004     23069     20517      3       3

Not many of my reads are removed as chimeric but almost all are removed from merging. My parameters for filtering were:
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE, compress=TRUE, multithread=TRUE)

I took out truncLen as I had previously tried truncLen=c(240,160) and got zero output, but I'm wondering if there is a better recommendation for that parameter? I'm also unsure of what truncLen does/what each number means in c(x,y). 

I am also wondering if I want to re-do the filtering step, can I simply enter the function again or do I have to delete my filtered files and re-start from the beginning? 

Thank you in advance!

**Comment by benjjneb** [comment]
Merging is clearly not working, usually that is caused by non-overlapping reads, but given what you have described that doesn't seem to be the case here.

What sequencing tech are you using, e.g. Miseq 2x250?
What is the output of `head(dadaFs[[1]]$clustering)` and `head(dadaRs[[2]]$clustering)`?

> I'm also unsure of what truncLen does/what each number means in c(x,y).

`truncLen` determines the position at which each read is truncated, with `x` being the truncation value for the forward reads, and `y` being for the reverse reads if you provide two values ala `truncLen=c(x,y)`.

> I am also wondering if I want to re-do the filtering step, can I simply enter the function again or do I have to delete my filtered files and re-start from the beginning?

The previous files will be automatically overwritten (with a message informing you of that fact).

**Comment by natasharg** [comment]
The run was performed using MiSeq Nano 2x150

> head(dadaFs[[1]]$clustering)
                                                                                                                                                 sequence
1 CCTACGGGTGGCTGCAGTGGGGAATTTTCCGCAATGGGCGAAAGCCTGACGGAGCAATGCCGCGTGAAGGCAGAAGGCCCACGGGTCATGAACTTCTTTTCTCGGAGAAGAAAAAATGACGGTATCTGAGGAATAAGCATCGGCTAACTCT
2 CCTACGGGGGGCTGCAGCAGGGAATCTTGCGCAATGGGCGAAAGCCTGACGCAGCAACGCCGCGTGGGGGAAGAAGGCCTTCGGGTTGTAAACCCCTTTCAGGAGGGAAGAAAATGACGGTACCTCCAGAAGAAGCACCGGCCAACTACGT
3 CCTACGGGTGGCTGCAGTGGGGAATATTGGACAATGGGCGAAAGCCTGATCCAGCAATGCCGCGTGTGTGAAGAAGGCCTGCGGGTTGTAAAGCACTTTCAGCAGGGAAGAAAAGCCTGGCGCTAATACCGCCGGGTCTTGACGTCACCTG
4 CCTACGGGGGGCTGCAGTAGGGAATATTGGTCAATGGGTGAGAGCCTGAACCAGCCATGCCGCGTGCAGGAAGACGGCCTTCTGGGTTGTAAACTGCTTTTACCAGGGGATAAAAAGATCTTGCGAGATTAATTGAAGGTACCTGGAGAAT
5 CCTACGGGTGGCTGCAGTGGGGAATATTGGACAATGGGGGCAACCCTGATCCAGCAATGCCGCGTGTGTGAAGAAGGCCTGCGGGTTGTAAAGCACTTTCAGAGGGGAAGAAAAAATCGGAGCTAATACCTCCGATCTTGACGTTACCCTT
6 CCTACGGGAGGCTGCAGTGGGGAATTTTGGGCAATGGCCGAAAGGCTGACCCAGCGACGCCGCGTGCGGGAGGACGCCCCTCGGGGTGTAAACCGCTGTCAGAGGGGACGAATCCCCGACGTTCGAACAGGGCGTCGGGTTGACGGTACCC
  abundance  n0 n1 nunq pval birth_type birth_pval   birth_fold birth_ham birth_qave
1       383 174 79  188    0          I         NA           NA        NA         NA
2       164  36 28  111    0          A          0 3.213217e+52        26   35.84615
3       304  19 46  219    0          A          0 8.181050e+85        40   36.27500
4       224  18 46  158    0          A          0 5.248302e+78        37   34.94595
5       433  19 39  351    0          A          0 2.292219e+31        14   36.21429
6       212  21 43  151    0          A          0 1.259043e+91        43   35.72093

> head(dadaRs[[2]]$clustering)
                                                                                                                                                 sequence
1 GACTACCGGGGTATCTAATCCTGTTTGCTCCCCACGCTTTCGCTCCTCAGCGTCAGTGCCGGCCCAGAGAACCGCCTTCGCTACTGGTGTTCCTCCCGATATCTGCGCATTCCACCGCTACACCGGGAATTCCATTCTCCTCTACCGGACT
2 GACTACAGGGGTATCTAATCCTGTTTGCTCCCCACGCTTTCGCACCTCAGCGTCAGTATCGGGCCAGTGAGCCGCCTTCGCCACTGGTGTTCTTGCGAATATCTACGAATTTCACCTCTACACTCGCAGTTCCACTCACCTCTCCCGAACT
3 GACTACCCGGGTATCTAATCCCGTTTGCTCCCCACGCTTTCGTGCCTGAGCGTCAGGACAGACCCAGCCACCTGTCTTCACCTTTGGCGTTCCGCTAGATCTCTACGCATTTCACCGCTCCACCTAGCGTTCCGATGGCCTCTATCTGCCT
4 GACTACCGGGGTATCTAATCCTGTTTGCTCCCCACGCTTTCGCGCCTCAGCGTCGGATCCGGCCCAGCGGGCCGCCTTCGCCACCGGTGTTCTTCCCGATATCTACGCATTCCACCGCTACACCGGGAATTCCACCCGCCCCTACCGGTCC
5 GACTACTCGGGTATCTAATCCCGTTCGCTCCCCACGCTTTCGCACATGTGCGTCAGGCCGATGCCAGGGCGCCGCTTTCGCCACTGGTATTCCTCCAGATATCTACGCATTCCACCACTACACCTGGAATTCTGCACCCCTCTCATCGCCT
6 GACTACTGGGGTATCTAATCCTGTTCGCTCCCCACGCTTTCGCTCCTCAGCGTCAGTATCGGCCCAGAGATCCGCCTTCGCCACCGGTGTTCCTCCTGATATCTGCGCATTTCACCGCTACACCAGGAATTCCGATCTCCCCTACCGAACT
  abundance n0 n1 nunq          pval birth_type    birth_pval   birth_fold birth_ham birth_qave
1      2368 19 66 2222  0.000000e+00          I            NA           NA        NA         NA
2       987  8 17  941  0.000000e+00          A  0.000000e+00 5.403713e+42        24   34.45833
3       286  4 10  274 7.160755e-271          A 1.871324e-198 2.908229e+67        37   33.35135
4       678  8 19  642 2.305892e-229          A 7.154029e-195 3.850506e+28        17   33.35294
5       367  5 13  350 3.879280e-242          A 7.701293e-189 1.848418e+48        27   34.33333
6      1193  8 36 1129 1.464937e-186          A 1.166489e-157 1.859819e+23        14   33.85714

Thanks for the help.

**Comment by benjjneb** [comment]
There's probably a couple things going on. First, you still have primers on your reads, in particular the 806R primer is present on the reverse reads. That should be trimmed off before using DADA2, or alternatively removed via the `trimLeft` argument in the filtering step.

Second, my suspicion is that a lot of this data might be spurious. When I BLAST your forward sequences I get best hits as things like `Pinus crassicorticea` and `Thiorhodococcus pfennigii`. When I BLAST those reverse sequences I get things like `Pseudolabrys sp.` and `Mexican potato purple top phytoplasma` as best hits (although some of the best hits are pretty far off). Does that seem like the kind of sequences you're expecting? My guess is that this PCR/sequencing didn't turn out very well.

**Comment by natasharg** [comment]
I see. I've trimmed the primers off the sequences now. Just checking, if the forward primer is 19 bp and reverse primer is 20 bp, it would be correct to use `trimLeft=c(19,20)`? 

Once I did that I got slightly higher output values after merging but still not a significant amount. 
`
           input filtered denoisedF denoisedR merged nonchim`
`Sample-1  29112    27578     25657     23383      6       6`
`Sample-2  19059    18144     17072     15521     10      10`
`Sample-3 99195    95285     93521     92410     34      34`
`Sample-4 76890    74315     73074     70903    315     315`
`Sample-5   29173    28156     26546     26283     14      14`
`Sample-6   26079    25180     23791     23044      6       6`

Could it also be due to my missing truncLen parameter? If my run was bad would it be expected that the drop after merging be this large? Thanks for the input, I'm definitely also considering that the run did not turn out well.

**Comment by benjjneb** [comment]
> I see. I've trimmed the primers off the sequences now. Just checking, if the forward primer is 19 bp and reverse primer is 20 bp, it would be correct to use trimLeft=c(19,20)?

Well, if both your fwd and rev primers are on your sequences, your reads barely overlap. The V4 region is ~252 nts long. Add 39 bases of primers and you are up to 291 nts. You only have 2x150 = 300 nts of total sequencing, so just ~9 nts of overlap. If you look at the documentation for `mergePairs` you'll see it requires 12nts of overlap by default.

You can try `mergePairs(..., minOverlap=5)` and see how that works. Perhaps even drop to `4` at the risk of a few false merges.

Going forward, I would recommend using a library prep strategy that doesn't sequence your primers if using 2x150 sequencing here though. It's better to have more overlap than the tiny amount you have in this setup.

**Comment by natasharg** [comment]
Got it. Thanks for the help. I added the 'minOverlap=4' and got more reads but not sure if they are completely accurate. Is there a way to use DADA2 to only take into account the forward reads?

**Comment by benjjneb** [comment]
> Is there a way to use DADA2 to only take into account the forward reads?

Sure, just run the tutorial the same, but skip the parts processing the reverse reads or merging. Then make the sequence table with the forward reads alone, i.e. `seqtab <- makeSequenceTable(dadaFs)`. Everything else is the same.

**Comment by dixi06** [comment]
Hello sir,

I am having trouble with dada2(qiime2) in merging. I am working on 16S data sequenced on Illumina Miseq 2x300bp using 515f&806r primers(V4 region). I have attached my quality check before denoise. I have used truncLen=c(150, 120)  which gave me a total reads 67% and then have used truncLen=c(250, 200) gave me a total 63%. Also truncLen=c(200,150) gave me similar. I am fairly new to this so I don't understand the logic behind it.  I am losing reads in merging. Is it ok to lose 40% reads? Does this trunclen make sense? Can you please help me with it?  


![Screen Shot 2019-05-28 at 1 41 26 PM](https://user-images.githubusercontent.com/21266374/58510791-5a10e680-814e-11e9-983a-b8f19b279e37.png)

Please let me know if you need more information. 


Thanks a lot,

Regards,

Dixi

**Comment by benjjneb** [comment]
@dixi06 You want to keep enough nucleotides so that the forward and reverse reads overlap sufficiently to merge, while removing low quality bases and any "overhang", i.e. bases that come from reading past the opposite primer.

Are your primers sequenced? If not, this amplicon is ~252 nts, in which case you want to keep 280+ total nucleotides, i.e. forward `truncLen` plus reverse `truncLen` greater than 280. You also want to truncate each read at 250 nts or less to avoid reading into the opposite primer and the adapter sequence after it.

Both `truncLen=c(250, 200)` and `truncLen=c(200,150)` would work for that purpose.

**Comment by hlfreund** [comment]
Hi Dr. Callahan,

I think I am having a similar issue as @natasharg. We sequenced 16S V3/V4 region with MiSeq Nano (2x250 bp). It appears that our primers are still attached within the initial ~30 NTs based on the FastQC reports.

I tried trimming the primers with bbduk and ran these sequences through DADA2, but was getting merged reads that were 240 - 300+ NTs long (so some overlap but a lot of variation in the number of overlaps). The sequences are fairly good quality before trimming, but I was hoping to get more merged reads that fit within the 240-260 NT range.

Would it be best to try and cut the first ~30 bp of the reads with the truncLen parameter? Or is it okay to move forward and include merged reads that are 300 NTs or more? I am just not sure how to proceed. Thank you for your insight!

**Comment by benjjneb** [comment]
@hlfreund Typically we see the best results by trimming the primers using the simple `trimLeft` approach (i.e. just cut off the fixed number of bases equal to the primer length), as long as you are using fixed length primers and no heterogeneity spacers. But I would look up your primer sequences and use the exact primer lengths for that (and can double check in the sequence files that the primers are at the start of the reads).

Second, the distribution of V3V4 regions in nature has a multi-modal distribution, with two modes that differ by ~30 nts in length (e.g. [Fig 2 here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4315398/figure/pone.0116955.g002/)). So factor that into your expectations about the merged length distribution.

**Comment by hlfreund** [comment]
Thank you for the tip regarding the trimLeft parameter, I will utilize this when I run the DADA2 pipeline again.

Also thank you for including that figure regarding the length distribution of the V3/V4 regions. I think my concern came about when I followed the tutorial and saw that there was a suggestion for restricting the merged sequences between a certain length. 

I also forgot to mention that I only ran the pipeline with 5 samples (to get preliminary data out for a presentation) - but plan on running the pipeline with 72 samples, so I am sure I will get more successful merged reads then.

Thank you for your quick correspondence, this great pipeline, and your wonderful tutorial.
